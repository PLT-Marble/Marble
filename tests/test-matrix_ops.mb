int function main() {
    // mat +/- mat
    matrix a = [1.1, 2.2; 3.3, 4.4];
    matrix b = [5.5, 6.6; 7.7, 8.8];
    matrix c = a + b;
    printmf(c);
    matrix d = a - b;
    printmf(d);
    // int * mat mat * int
    int d = 3;
    matrix e = d * a;
    printmf(e);
    matrix f = a * d;
    printmf(f);
    // float * mat mat * float
    float g = 1.1;
    matrix h = g * a;
    printmf(h);
    matrix i = b * g;
    printmf(i);
    // mat * mat
    matrix j = a * b;
    printmf(j);
    matrix k = b * a;
    printmf(k);
    return 0;
}