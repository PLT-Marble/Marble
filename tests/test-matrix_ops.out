6.600000 8.800000
11.000000 13.200000
-4.400000 -4.400000
-4.400000 -4.400000
3.300000 6.600000
9.900000 13.200000
3.300000 6.600000
9.900000 13.200000
1.210000 2.420000
3.630000 4.840000
6.050000 7.260000
8.470000 9.680000
22.990000 26.620000
52.030000 60.500000
27.830000 41.140000
37.510000 55.660000
